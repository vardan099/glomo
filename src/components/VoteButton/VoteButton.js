import React from 'react';
import PropTypes from "prop-types";

const VoteButton = (props) => {
    const {buttonText, clickHandler, voteType, gameId, type} = props;
    return (
        <button data-test="voteButton" className={`vote-button bg-${type}`} onClick={() => clickHandler(gameId, voteType)}>{buttonText}</button>
    )
};

VoteButton.propTypes = {
    buttonText: PropTypes.string,
    clickHandler: PropTypes.func,
    voteType: PropTypes.number,
    type: PropTypes.string,
    gameId: PropTypes.string
};


export default VoteButton;