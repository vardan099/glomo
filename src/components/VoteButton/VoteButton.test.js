import React from 'react';
import { shallow } from 'enzyme';
import { checkProps } from '../../TestUtils/index';
import VoteButton from './VoteButton';

describe('VoteButton Component', () => {

    describe('Checking PropTypes', () => {
        it('Should NOT throw a warning', () => {
            const expectedProps = {
                buttonText: jest.string,
                clickHandler: jest.func,
                voteType: jest.number,
                gameId: jest.string,
                type: jest.string,
            };
            const propsError = checkProps(VoteButton, expectedProps);
            expect(propsError).toBeUndefined();
        });

    });

    describe('Component Renders', () => {
        let wrapper;
        const props = {
            buttonText: "Home Team",
            clickHandler: jest.fn(),
            voteType: 2,
            gameId: "CSJn3kZhdx",
            type: "home"
        };

        beforeEach(() => {
            wrapper = shallow(<VoteButton {...props} />);
        });

        it('renders correctly', () => {
            expect(wrapper).toMatchSnapshot();
        });

        it('Test click event', () => {
            wrapper.find('button').simulate('click');
            expect(props.clickHandler.mock.calls.length).toEqual(1);
        });
    });


});