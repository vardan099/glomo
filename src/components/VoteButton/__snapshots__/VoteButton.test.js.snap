// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VoteButton Component Component Renders renders correctly 1`] = `ShallowWrapper {}`;
